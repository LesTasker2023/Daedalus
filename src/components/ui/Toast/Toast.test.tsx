/**
 * ðŸ§ª Toast Component Test Suite
 *
 * Comprehensive test coverage for Toast notification system following
 * TriggerKings enterprise testing standards.
 */

import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import React from 'react';
import { beforeEach, describe, expect, it, vi } from 'vitest';
import {
  Toast,
  ToastAction,
  ToastClose,
  ToastDescription,
  ToastProvider,
  ToastTitle,
  ToastViewport,
} from './toast';
import { Toaster } from './toaster';
import { useToast } from './use-toast';

// ===== TEST UTILITIES =====

function TestWrapper({ children }: { children: React.ReactNode }) {
  return (
    <ToastProvider>
      {children}
      <ToastViewport />
    </ToastProvider>
  );
}

function HookTestComponent() {
  const { toast } = useToast();

  return (
    <div>
      <button
        onClick={() => toast({ title: 'Default Toast', description: 'Default description' })}
        data-testid="trigger-default"
      >
        Default
      </button>
      <button
        onClick={() =>
          toast({ title: 'Success!', description: 'Success message', variant: 'success' })
        }
        data-testid="trigger-success"
      >
        Success
      </button>
      <button
        onClick={() => toast({ title: 'Error!', description: 'Error message', variant: 'error' })}
        data-testid="trigger-error"
      >
        Error
      </button>
    </div>
  );
}

// ===== COMPONENT TESTS =====

describe('Toast Component', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  describe('Toast Rendering', () => {
    it('renders basic toast with title and description', () => {
      render(
        <TestWrapper>
          <Toast data-testid="toast-root">
            <ToastTitle data-testid="toast-title">Test Title</ToastTitle>
            <ToastDescription data-testid="toast-description">Test Description</ToastDescription>
            <ToastClose data-testid="toast-close" />
          </Toast>
        </TestWrapper>
      );

      expect(screen.getByTestId('toast-root')).toBeInTheDocument();
      expect(screen.getByTestId('toast-title')).toBeInTheDocument();
      expect(screen.getByTestId('toast-description')).toBeInTheDocument();
      expect(screen.getByTestId('toast-close')).toBeInTheDocument();
      expect(screen.getByText('Test Title')).toBeInTheDocument();
      expect(screen.getByText('Test Description')).toBeInTheDocument();
    });

    it('renders toast with only title', () => {
      render(
        <TestWrapper>
          <Toast data-testid="toast-root">
            <ToastTitle data-testid="toast-title">Only Title</ToastTitle>
            <ToastClose />
          </Toast>
        </TestWrapper>
      );

      expect(screen.getByTestId('toast-root')).toBeInTheDocument();
      expect(screen.getByText('Only Title')).toBeInTheDocument();
      expect(screen.queryByTestId('toast-description')).not.toBeInTheDocument();
    });
  });

  describe('Toast Variants', () => {
    const variants = ['default', 'success', 'error', 'warning', 'info'] as const;

    variants.forEach(variant => {
      it(`renders ${variant} variant with correct styling`, () => {
        render(
          <TestWrapper>
            <Toast variant={variant} data-testid={`toast-${variant}`}>
              <ToastTitle>{`${variant} Title`}</ToastTitle>
              <ToastClose />
            </Toast>
          </TestWrapper>
        );

        const toast = screen.getByTestId(`toast-${variant}`);
        expect(toast).toBeInTheDocument();
        expect(screen.getByText(`${variant} Title`)).toBeInTheDocument();

        if (variant !== 'default') {
          expect(toast.className).toContain('bg-');
        }
      });
    });
  });

  describe('Toast Actions', () => {
    it('renders and handles action button clicks', () => {
      const mockAction = vi.fn();

      render(
        <TestWrapper>
          <Toast data-testid="toast-root">
            <ToastTitle>Actionable Toast</ToastTitle>
            <ToastAction onClick={mockAction} data-testid="toast-action" altText="Action Button">
              Click Me
            </ToastAction>
            <ToastClose />
          </Toast>
        </TestWrapper>
      );

      const actionButton = screen.getByTestId('toast-action');
      expect(actionButton).toBeInTheDocument();
      expect(screen.getByText('Click Me')).toBeInTheDocument();

      fireEvent.click(actionButton);
      expect(mockAction).toHaveBeenCalledTimes(1);
    });
  });

  describe('useToast Hook', () => {
    it('creates default toast via hook', async () => {
      render(
        <TestWrapper>
          <HookTestComponent />
          <Toaster />
        </TestWrapper>
      );

      fireEvent.click(screen.getByTestId('trigger-default'));

      await waitFor(
        () => {
          expect(screen.getByText('Default Toast')).toBeInTheDocument();
          expect(screen.getByText('Default description')).toBeInTheDocument();
        },
        { timeout: 3000 }
      );
    });

    it('creates success toast with correct styling', async () => {
      render(
        <TestWrapper>
          <HookTestComponent />
          <Toaster />
        </TestWrapper>
      );

      fireEvent.click(screen.getByTestId('trigger-success'));

      await waitFor(() => {
        expect(screen.getByText('Success!')).toBeInTheDocument();
        expect(screen.getByText('Success message')).toBeInTheDocument();
      });
    });

    it('handles multiple toasts simultaneously', async () => {
      render(
        <TestWrapper>
          <HookTestComponent />
          <Toaster />
        </TestWrapper>
      );

      // Test that error toast appears first
      fireEvent.click(screen.getByTestId('trigger-error'));

      await waitFor(() => {
        expect(screen.getByText('Error!')).toBeInTheDocument();
        expect(screen.getByText('Error message')).toBeInTheDocument();
      });

      // Test that we can create another toast
      fireEvent.click(screen.getByTestId('trigger-success'));

      await waitFor(() => {
        // At least one toast should be visible
        expect(screen.getByText('Success!')).toBeInTheDocument();
      });
    });
  });

  describe('Accessibility', () => {
    it('has proper semantic structure', () => {
      render(
        <TestWrapper>
          <Toast data-testid="accessible-toast">
            <ToastTitle>Accessible Title</ToastTitle>
            <ToastDescription>Accessible Description</ToastDescription>
            <ToastClose />
          </Toast>
        </TestWrapper>
      );

      const toast = screen.getByTestId('accessible-toast');
      expect(toast).toBeInTheDocument();
      expect(screen.getByText('Accessible Title')).toBeInTheDocument();
      expect(screen.getByText('Accessible Description')).toBeInTheDocument();
    });

    it('supports keyboard navigation', () => {
      render(
        <TestWrapper>
          <Toast data-testid="keyboard-toast">
            <ToastTitle>Keyboard Navigation</ToastTitle>
            <ToastClose data-testid="keyboard-close" />
          </Toast>
        </TestWrapper>
      );

      const closeButton = screen.getByTestId('keyboard-close');
      closeButton.focus();
      expect(document.activeElement).toBe(closeButton);
    });
  });

  describe('Toast Lifecycle', () => {
    it('handles onOpenChange callback', () => {
      const mockOpenChange = vi.fn();

      render(
        <TestWrapper>
          <Toast onOpenChange={mockOpenChange} data-testid="lifecycle-toast">
            <ToastTitle>Lifecycle Test</ToastTitle>
            <ToastClose data-testid="lifecycle-close" />
          </Toast>
        </TestWrapper>
      );

      fireEvent.click(screen.getByTestId('lifecycle-close'));
      expect(mockOpenChange).toHaveBeenCalledWith(false);
    });
  });

  describe('Component Props', () => {
    it('accepts and applies custom className', () => {
      render(
        <TestWrapper>
          <Toast className="custom-toast-class" data-testid="custom-class-toast">
            <ToastTitle>Custom Class Test</ToastTitle>
            <ToastClose />
          </Toast>
        </TestWrapper>
      );

      const toast = screen.getByTestId('custom-class-toast');
      expect(toast).toHaveClass('custom-toast-class');
    });

    it('forwards ref correctly', () => {
      const ref = React.createRef<HTMLLIElement>();

      render(
        <TestWrapper>
          <Toast ref={ref} data-testid="ref-toast">
            <ToastTitle>Ref Test</ToastTitle>
            <ToastClose />
          </Toast>
        </TestWrapper>
      );

      expect(ref.current).toBeInstanceOf(HTMLElement);
    });
  });

  describe('Error Handling', () => {
    it('handles empty content gracefully', () => {
      const consoleSpy = vi.spyOn(console, 'error').mockImplementation(() => {});

      render(
        <TestWrapper>
          <Toast data-testid="empty-toast">
            <ToastTitle></ToastTitle>
            <ToastClose />
          </Toast>
        </TestWrapper>
      );

      expect(screen.getByTestId('empty-toast')).toBeInTheDocument();
      consoleSpy.mockRestore();
    });
  });

  describe('Snapshots', () => {
    it('matches default toast snapshot', () => {
      const { container } = render(
        <TestWrapper>
          <Toast data-testid="default-toast">
            <ToastTitle>Default Toast</ToastTitle>
            <ToastDescription>This is a default toast message</ToastDescription>
            <ToastClose />
          </Toast>
        </TestWrapper>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches all variants snapshot', () => {
      const { container } = render(
        <TestWrapper>
          <Toast variant="default" data-testid="variant-default">
            <ToastTitle>Default</ToastTitle>
            <ToastDescription>Default variant</ToastDescription>
          </Toast>
          <Toast variant="success" data-testid="variant-success">
            <ToastTitle>Success</ToastTitle>
            <ToastDescription>Success variant</ToastDescription>
          </Toast>
          <Toast variant="error" data-testid="variant-error">
            <ToastTitle>Error</ToastTitle>
            <ToastDescription>Error variant</ToastDescription>
          </Toast>
          <Toast variant="warning" data-testid="variant-warning">
            <ToastTitle>Warning</ToastTitle>
            <ToastDescription>Warning variant</ToastDescription>
          </Toast>
          <Toast variant="info" data-testid="variant-info">
            <ToastTitle>Info</ToastTitle>
            <ToastDescription>Info variant</ToastDescription>
          </Toast>
        </TestWrapper>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches toast with action snapshot', () => {
      const { container } = render(
        <TestWrapper>
          <Toast data-testid="toast-with-action">
            <ToastTitle>Toast with Action</ToastTitle>
            <ToastDescription>This toast has an action button</ToastDescription>
            <ToastAction altText="Undo action">Undo</ToastAction>
            <ToastClose />
          </Toast>
        </TestWrapper>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches toast provider and viewport snapshot', () => {
      const { container } = render(
        <ToastProvider>
          <div>Content</div>
          <ToastViewport data-testid="toast-viewport" />
        </ToastProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches toaster component snapshot', () => {
      const { container } = render(<Toaster />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
