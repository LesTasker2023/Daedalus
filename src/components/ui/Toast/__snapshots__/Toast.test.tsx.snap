// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Toast Component > Snapshots > matches all variants snapshot 1`] = `
<div
  aria-label="Notifications (F8)"
  role="region"
  style=""
  tabindex="-1"
>
  <span
    style="position: fixed; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
    tabindex="0"
  />
  <ol
    class="fixed top-0 z-[100] flex max-h-screen w-full flex-col-reverse p-4 sm:bottom-0 sm:right-0 sm:top-auto sm:flex-col md:max-w-[420px]"
    tabindex="-1"
  >
    <li
      class="group pointer-events-auto relative flex w-full items-center justify-between space-x-4 overflow-hidden rounded-md p-6 pr-8 shadow-lg transition-all data-[swipe=cancel]:translate-x-0 data-[swipe=end]:translate-x-[var(--radix-toast-swipe-end-x)] data-[swipe=move]:translate-x-[var(--radix-toast-swipe-move-x)] data-[swipe=move]:transition-none data-[state=open]:animate-in data-[state=closed]:animate-out data-[swipe=end]:animate-out data-[state=closed]:fade-out-80 data-[state=closed]:slide-out-to-right-full data-[state=open]:slide-in-from-top-full data-[state=open]:sm:slide-in-from-bottom-full border bg-background text-foreground"
      data-radix-collection-item=""
      data-state="open"
      data-swipe-direction="right"
      data-testid="variant-default"
      style="user-select: none;"
      tabindex="0"
    >
      <div
        class="text-sm font-semibold"
      >
        Default
      </div>
      <div
        class="text-sm opacity-90"
      >
        Default variant
      </div>
    </li>
    <li
      class="group pointer-events-auto relative flex w-full items-center justify-between space-x-4 overflow-hidden rounded-md border p-6 pr-8 shadow-lg transition-all data-[swipe=cancel]:translate-x-0 data-[swipe=end]:translate-x-[var(--radix-toast-swipe-end-x)] data-[swipe=move]:translate-x-[var(--radix-toast-swipe-move-x)] data-[swipe=move]:transition-none data-[state=open]:animate-in data-[state=closed]:animate-out data-[swipe=end]:animate-out data-[state=closed]:fade-out-80 data-[state=closed]:slide-out-to-right-full data-[state=open]:slide-in-from-top-full data-[state=open]:sm:slide-in-from-bottom-full border-green-200 bg-green-50 text-green-800 dark:border-green-800 dark:bg-green-900 dark:text-green-100"
      data-radix-collection-item=""
      data-state="open"
      data-swipe-direction="right"
      data-testid="variant-success"
      style="user-select: none;"
      tabindex="0"
    >
      <div
        class="text-sm font-semibold"
      >
        Success
      </div>
      <div
        class="text-sm opacity-90"
      >
        Success variant
      </div>
    </li>
    <li
      class="group pointer-events-auto relative flex w-full items-center justify-between space-x-4 overflow-hidden rounded-md border p-6 pr-8 shadow-lg transition-all data-[swipe=cancel]:translate-x-0 data-[swipe=end]:translate-x-[var(--radix-toast-swipe-end-x)] data-[swipe=move]:translate-x-[var(--radix-toast-swipe-move-x)] data-[swipe=move]:transition-none data-[state=open]:animate-in data-[state=closed]:animate-out data-[swipe=end]:animate-out data-[state=closed]:fade-out-80 data-[state=closed]:slide-out-to-right-full data-[state=open]:slide-in-from-top-full data-[state=open]:sm:slide-in-from-bottom-full border-red-200 bg-red-50 text-red-800 dark:border-red-800 dark:bg-red-900 dark:text-red-100"
      data-radix-collection-item=""
      data-state="open"
      data-swipe-direction="right"
      data-testid="variant-error"
      style="user-select: none;"
      tabindex="0"
    >
      <div
        class="text-sm font-semibold"
      >
        Error
      </div>
      <div
        class="text-sm opacity-90"
      >
        Error variant
      </div>
    </li>
    <li
      class="group pointer-events-auto relative flex w-full items-center justify-between space-x-4 overflow-hidden rounded-md border p-6 pr-8 shadow-lg transition-all data-[swipe=cancel]:translate-x-0 data-[swipe=end]:translate-x-[var(--radix-toast-swipe-end-x)] data-[swipe=move]:translate-x-[var(--radix-toast-swipe-move-x)] data-[swipe=move]:transition-none data-[state=open]:animate-in data-[state=closed]:animate-out data-[swipe=end]:animate-out data-[state=closed]:fade-out-80 data-[state=closed]:slide-out-to-right-full data-[state=open]:slide-in-from-top-full data-[state=open]:sm:slide-in-from-bottom-full border-yellow-200 bg-yellow-50 text-yellow-800 dark:border-yellow-800 dark:bg-yellow-900 dark:text-yellow-100"
      data-radix-collection-item=""
      data-state="open"
      data-swipe-direction="right"
      data-testid="variant-warning"
      style="user-select: none;"
      tabindex="0"
    >
      <div
        class="text-sm font-semibold"
      >
        Warning
      </div>
      <div
        class="text-sm opacity-90"
      >
        Warning variant
      </div>
    </li>
    <li
      class="group pointer-events-auto relative flex w-full items-center justify-between space-x-4 overflow-hidden rounded-md border p-6 pr-8 shadow-lg transition-all data-[swipe=cancel]:translate-x-0 data-[swipe=end]:translate-x-[var(--radix-toast-swipe-end-x)] data-[swipe=move]:translate-x-[var(--radix-toast-swipe-move-x)] data-[swipe=move]:transition-none data-[state=open]:animate-in data-[state=closed]:animate-out data-[swipe=end]:animate-out data-[state=closed]:fade-out-80 data-[state=closed]:slide-out-to-right-full data-[state=open]:slide-in-from-top-full data-[state=open]:sm:slide-in-from-bottom-full border-blue-200 bg-blue-50 text-blue-800 dark:border-blue-800 dark:bg-blue-900 dark:text-blue-100"
      data-radix-collection-item=""
      data-state="open"
      data-swipe-direction="right"
      data-testid="variant-info"
      style="user-select: none;"
      tabindex="0"
    >
      <div
        class="text-sm font-semibold"
      >
        Info
      </div>
      <div
        class="text-sm opacity-90"
      >
        Info variant
      </div>
    </li>
  </ol>
  <span
    style="position: fixed; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
    tabindex="0"
  />
</div>
`;

exports[`Toast Component > Snapshots > matches default toast snapshot 1`] = `
<div
  aria-label="Notifications (F8)"
  role="region"
  style=""
  tabindex="-1"
>
  <span
    style="position: fixed; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
    tabindex="0"
  />
  <ol
    class="fixed top-0 z-[100] flex max-h-screen w-full flex-col-reverse p-4 sm:bottom-0 sm:right-0 sm:top-auto sm:flex-col md:max-w-[420px]"
    tabindex="-1"
  >
    <li
      class="group pointer-events-auto relative flex w-full items-center justify-between space-x-4 overflow-hidden rounded-md p-6 pr-8 shadow-lg transition-all data-[swipe=cancel]:translate-x-0 data-[swipe=end]:translate-x-[var(--radix-toast-swipe-end-x)] data-[swipe=move]:translate-x-[var(--radix-toast-swipe-move-x)] data-[swipe=move]:transition-none data-[state=open]:animate-in data-[state=closed]:animate-out data-[swipe=end]:animate-out data-[state=closed]:fade-out-80 data-[state=closed]:slide-out-to-right-full data-[state=open]:slide-in-from-top-full data-[state=open]:sm:slide-in-from-bottom-full border bg-background text-foreground"
      data-radix-collection-item=""
      data-state="open"
      data-swipe-direction="right"
      data-testid="default-toast"
      style="user-select: none;"
      tabindex="0"
    >
      <div
        class="text-sm font-semibold"
      >
        Default Toast
      </div>
      <div
        class="text-sm opacity-90"
      >
        This is a default toast message
      </div>
      <button
        class="absolute right-2 top-2 rounded-md p-1 text-foreground/50 opacity-0 transition-opacity hover:text-foreground focus:opacity-100 focus:outline-none focus:ring-2 group-hover:opacity-100 group-[.success]:text-green-600 group-[.error]:text-red-600 group-[.warning]:text-yellow-600 group-[.info]:text-blue-600"
        data-radix-toast-announce-exclude=""
        toast-close=""
        type="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-x h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 6 6 18"
          />
          <path
            d="m6 6 12 12"
          />
        </svg>
      </button>
    </li>
  </ol>
  <span
    style="position: fixed; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
    tabindex="0"
  />
</div>
`;

exports[`Toast Component > Snapshots > matches toast provider and viewport snapshot 1`] = `
<div>
  Content
</div>
`;

exports[`Toast Component > Snapshots > matches toast with action snapshot 1`] = `
<div
  aria-label="Notifications (F8)"
  role="region"
  style=""
  tabindex="-1"
>
  <span
    style="position: fixed; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
    tabindex="0"
  />
  <ol
    class="fixed top-0 z-[100] flex max-h-screen w-full flex-col-reverse p-4 sm:bottom-0 sm:right-0 sm:top-auto sm:flex-col md:max-w-[420px]"
    tabindex="-1"
  >
    <li
      class="group pointer-events-auto relative flex w-full items-center justify-between space-x-4 overflow-hidden rounded-md p-6 pr-8 shadow-lg transition-all data-[swipe=cancel]:translate-x-0 data-[swipe=end]:translate-x-[var(--radix-toast-swipe-end-x)] data-[swipe=move]:translate-x-[var(--radix-toast-swipe-move-x)] data-[swipe=move]:transition-none data-[state=open]:animate-in data-[state=closed]:animate-out data-[swipe=end]:animate-out data-[state=closed]:fade-out-80 data-[state=closed]:slide-out-to-right-full data-[state=open]:slide-in-from-top-full data-[state=open]:sm:slide-in-from-bottom-full border bg-background text-foreground"
      data-radix-collection-item=""
      data-state="open"
      data-swipe-direction="right"
      data-testid="toast-with-action"
      style="user-select: none;"
      tabindex="0"
    >
      <div
        class="text-sm font-semibold"
      >
        Toast with Action
      </div>
      <div
        class="text-sm opacity-90"
      >
        This toast has an action button
      </div>
      <button
        class="inline-flex h-8 shrink-0 items-center justify-center rounded-md border bg-transparent px-3 text-xs font-medium ring-offset-background transition-colors hover:bg-secondary focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 group-[.success]:border-green-300 group-[.success]:hover:border-green-400 group-[.success]:hover:bg-green-100 group-[.error]:border-red-300 group-[.error]:hover:border-red-400 group-[.error]:hover:bg-red-100"
        data-radix-toast-announce-alt="Undo action"
        data-radix-toast-announce-exclude=""
        type="button"
      >
        Undo
      </button>
      <button
        class="absolute right-2 top-2 rounded-md p-1 text-foreground/50 opacity-0 transition-opacity hover:text-foreground focus:opacity-100 focus:outline-none focus:ring-2 group-hover:opacity-100 group-[.success]:text-green-600 group-[.error]:text-red-600 group-[.warning]:text-yellow-600 group-[.info]:text-blue-600"
        data-radix-toast-announce-exclude=""
        toast-close=""
        type="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-x h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 6 6 18"
          />
          <path
            d="m6 6 12 12"
          />
        </svg>
      </button>
    </li>
  </ol>
  <span
    style="position: fixed; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
    tabindex="0"
  />
</div>
`;

exports[`Toast Component > Snapshots > matches toaster component snapshot 1`] = `
<div
  aria-label="Notifications (F8)"
  role="region"
  style=""
  tabindex="-1"
>
  <span
    style="position: fixed; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
    tabindex="0"
  />
  <ol
    class="fixed top-0 z-[100] flex max-h-screen w-full flex-col-reverse p-4 sm:bottom-0 sm:right-0 sm:top-auto sm:flex-col md:max-w-[420px]"
    tabindex="-1"
  >
    <li
      class="group pointer-events-auto relative flex w-full items-center justify-between space-x-4 overflow-hidden rounded-md border p-6 pr-8 shadow-lg transition-all data-[swipe=cancel]:translate-x-0 data-[swipe=end]:translate-x-[var(--radix-toast-swipe-end-x)] data-[swipe=move]:translate-x-[var(--radix-toast-swipe-move-x)] data-[swipe=move]:transition-none data-[state=open]:animate-in data-[state=closed]:animate-out data-[swipe=end]:animate-out data-[state=closed]:fade-out-80 data-[state=closed]:slide-out-to-right-full data-[state=open]:slide-in-from-top-full data-[state=open]:sm:slide-in-from-bottom-full border-green-200 bg-green-50 text-green-800 dark:border-green-800 dark:bg-green-900 dark:text-green-100"
      data-radix-collection-item=""
      data-state="open"
      data-swipe-direction="right"
      style="user-select: none;"
      tabindex="0"
    >
      <div
        class="grid gap-1"
      >
        <div
          class="text-sm font-semibold"
        >
          Success!
        </div>
        <div
          class="text-sm opacity-90"
        >
          Success message
        </div>
      </div>
      <button
        class="absolute right-2 top-2 rounded-md p-1 text-foreground/50 opacity-0 transition-opacity hover:text-foreground focus:opacity-100 focus:outline-none focus:ring-2 group-hover:opacity-100 group-[.success]:text-green-600 group-[.error]:text-red-600 group-[.warning]:text-yellow-600 group-[.info]:text-blue-600"
        data-radix-toast-announce-exclude=""
        toast-close=""
        type="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-x h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 6 6 18"
          />
          <path
            d="m6 6 12 12"
          />
        </svg>
      </button>
    </li>
  </ol>
  <span
    style="position: fixed; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
    tabindex="0"
  />
</div>
`;
